PIL/GifImagePlugin.py

```diff
@@ -439,13 +439,9 @@ def _get_local_header(fp, im, offset, flags):
 
             # adjust the transparency index after optimize
             if len(used_palette_colors) < 256:
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+                transparent_color_exists = transparency in used_palette_colors
+                if transparent_color_exists:
+                    transparency = used_palette_colors.index(transparency)
```

Comment:
I have updated the PR, correcting a difference in flow. From what I can see, the lines are now covered by tests.